src/EntryPoints/RulesContentHandler.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\Rules\EntryPoints;
+
+use MediaWiki\Content\Content;
+use MediaWiki\Content\Renderer\ContentParseParams;
+use MediaWiki\Content\JsonContentHandler;
+use MediaWiki\Parser\ParserOutput;
+use MediaWiki\Title\Title;
+use ProfessionalWiki\Rules\RulesExtension;
+
+class RulesContentHandler extends JsonContentHandler {
+
+	public function __construct( $modelId = RulesExtension::RULES_CONTENT_MODEL ) {
+		parent::__construct( $modelId );
+	}
+
+	protected function getContentClass(): string {
+		return RulesContent::class;
+	}
+
+	public function isParserCacheSupported(): bool {
+		return true;
+	}
+
+	public function makeEmptyContent(): RulesContent {
+		return new RulesContent( RulesExtension::RULES_DEFAULT_CONFIG );
+	}
+
+	public function canBeUsedOn( Title $title ): bool {
+		return RulesExtension::getInstance()->isRulesPage( $title );
+	}
+
+	public function supportsDirectEditing(): bool {
+		// TODO: Need to decide if we want to allow direct editing
```

Comment:
Returning `false` will block users from editing the page with the edit action.
I am not sure whether we should allow it or not.